From 6d66283c5ecb4182f1951322628e5ce7e565c800 Mon Sep 17 00:00:00 2001
From: Kyle Harrison <khwebmail@gmail.com>
Date: Thu, 16 Jan 2014 17:28:46 +0000
Subject: [PATCH] Slikat build patch

---
 include/android/configuration.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/include/android/configuration.h b/include/android/configuration.h
index 6d8784d..bee1afb 100644
--- a/include/android/configuration.h
+++ b/include/android/configuration.h
@@ -77,6 +77,11 @@ enum {
     ACONFIGURATION_SCREENLONG_NO = 0x1,
     ACONFIGURATION_SCREENLONG_YES = 0x2,
 
+    ACONFIGURATION_UI_THEME_MODE_ANY = 0x0000,
+    ACONFIGURATION_UI_THEME_MODE_NORMAL = 0x0001,
+    ACONFIGURATION_UI_THEME_MODE_HOLO_DARK = 0x0002,
+    ACONFIGURATION_UI_THEME_MODE_HOLO_LIGHT = 0x0003,
+
     ACONFIGURATION_UI_MODE_TYPE_ANY = 0x00,
     ACONFIGURATION_UI_MODE_TYPE_NORMAL = 0x01,
     ACONFIGURATION_UI_MODE_TYPE_DESK = 0x02,
@@ -110,6 +115,7 @@ enum {
     ACONFIGURATION_SCREEN_SIZE = 0x0200,
     ACONFIGURATION_VERSION = 0x0400,
     ACONFIGURATION_SCREEN_LAYOUT = 0x0800,
+    ACONFIGURATION_UI_THEME_MODE = 0x0900,
     ACONFIGURATION_UI_MODE = 0x1000,
     ACONFIGURATION_SMALLEST_SCREEN_SIZE = 0x2000,
     ACONFIGURATION_LAYOUTDIR = 0x4000,
@@ -286,6 +292,16 @@ int32_t AConfiguration_getScreenLong(AConfiguration* config);
 void AConfiguration_setScreenLong(AConfiguration* config, int32_t screenLong);
 
 /**
+ * Return the current ACONFIGURATION_UI_THEME_MODE_* set in the configuration.
+ */
+int32_t AConfiguration_getUiThemeMode(AConfiguration* config);
+
+/**
+ * Set the current theme UI mode in the configuration.
+ */
+void AConfiguration_setUiThemeMode(AConfiguration* config, int32_t uiThemeMode);
+
+/**
  * Return the current ACONFIGURATION_UI_MODE_TYPE_* set in the configuration.
  */
 int32_t AConfiguration_getUiModeType(AConfiguration* config);
-- 
1.8.5.2

